source/dub/compilers/compiler.d

```diff
@@ -8,21 +8,14 @@
 module dub.compilers.compiler;
 
 public import dub.compilers.buildsettings;
-public import dub.dependency : Dependency;
```

Comment:
This is technically a breaking change I guess, which according to CONTRIBUTING.md cannot go into master. However, this line could be removed without adding any other line into the dub source itself, as no files that import `dub.compilers.compiler` use `Dependency`. Can a public import be deprecated?

Let me know if this line should be left as is.

Reply:
removing public imports should be considered a no-go. There are applications (most often IDEs/plugins) which depend on dub as a library and might have imported dub.compilers.compiler and used Dependency from it.

Deprecating public imports seems to work exactly the way you expect it to and should be done here instead (with a message telling you to import dub.dependency)